test_functions.TestTotalVariation sensitivie to numerical error

https://github.com/SyneRBI/SIRF-SuperBuild/actions/runs/9569076152/job/26380772313#step:14:1316
```
======================================================================
FAIL: test_get_p2_with_warm_start (test_functions.TestTotalVariation)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/SIRF-SuperBuild/SIRF-SuperBuild/build/sources/CIL/Wrappers/Python/test/test_functions.py", line 1547, in test_get_p2_with_warm_start
    np.testing.assert_almost_equal(np.sum(np.linalg.norm(test)),126.3372581, err_msg="Incorrect value of the proximal")
  File "/opt/hostedtoolcache/Python/3.10.12/x64/lib/python3.10/contextlib.py", line 79, in inner
    return func(*args, **kwds)
  File "/home/runner/virtualenv/lib/python3.10/site-packages/numpy/testing/_private/utils.py", line 589, in assert_almost_equal
    raise AssertionError(_build_err_msg())
AssertionError: 
Arrays are not almost equal to 7 decimals Incorrect value of the proximal
 ACTUAL: np.float64(126.33725735545158)
 DESIRED: 126.3372581
```

--- a/Wrappers/Python/cil/io/NEXUSDataReader.py
+++ b/Wrappers/Python/cil/io/NEXUSDataReader.py
@@ -100,7 +100,7 @@ class NEXUSDataReader(object):
 
         with h5py.File(self.file_name,'r') as dfile:
 
-            if np.string_(dfile.attrs['creator']) != np.string_('NEXUSDataWriter.py'):
+            if np.bytes_(dfile.attrs['creator']) != np.bytes_('NEXUSDataWriter.py'):
                 raise Exception('We can parse only files created by NEXUSDataWriter.py')
 
             ds_data = dfile['entry1/tomo_entry/data/data']
@@ -349,7 +349,7 @@ class NEXUSDataReader(object):
         #return ds_data.attrs.__contains__('geom_type')
         with h5py.File(self.file_name,'r') as dfile:
 
-            if np.string_(dfile.attrs['creator']) != np.string_('NEXUSDataWriter.py'):
+            if np.bytes_(dfile.attrs['creator']) != np.bytes_('NEXUSDataWriter.py'):
                 raise Exception('We can parse only files created by NEXUSDataWriter.py')
 
             ds_data = dfile['entry1/tomo_entry/data/data']

--- a/Wrappers/Python/cil/io/NEXUSDataWriter.py
+++ b/Wrappers/Python/cil/io/NEXUSDataWriter.py
@@ -123,7 +123,7 @@ class NEXUSDataWriter(object):
             f.attrs['file_name'] = self.file_name
             f.attrs['cil_version'] = version
             f.attrs['file_time'] = str(datetime.datetime.utcnow())
-            f.attrs['creator'] = np.string_('NEXUSDataWriter.py')
+            f.attrs['creator'] = np.bytes_('NEXUSDataWriter.py')
             f.attrs['NeXus_version'] = '4.3.0'
             f.attrs['HDF5_Version'] = h5py.version.hdf5_version
             f.attrs['h5py_version'] = h5py.version.version

